src/Sentry/HubExtensions.cs

```diff
@@ -244,4 +244,11 @@ public LockedScope(IHub hub)
         var transaction = hub.GetTransaction();
         return transaction?.IsSampled == true ? transaction : null;
     }
+
+    internal static Hub? GetRealHub(this IHub hub) => hub switch
+    {
+        Hub thisHub => thisHub,
+        HubAdapter when SentrySdk.CurrentHub is Hub sdkHub => sdkHub,
+        _ => null
+    };
```

Comment:
The only way to avoid this would be to add `IHub.StartSpan`... so it would be public (not internal - `IMetricHub` is internal remember).

Technically that would be a breaking change (adding a method to an interface) although I seriously doubt many of our customers are implementing their own `IHub` so not sure how strict we want to be on that. 

If we do add a method to that interface, I think we want to be very comfortable with what the method signature should look like since at that point, we definitely can't change it without a major version bump.

Reply:
OK, I found another way to work around this by converting it from an interface method to an extension method... the implementation only uses methods that are part of `IHub` anyway, so this works. 

Also, so that that extension is available in the MetricAggregator, I've changed `IMetricHub` to extend `IHub` (so the same extension is available there too).